site/src/modules/resources/AgentLogs/useAgentLogs.ts

```diff
@@ -15,22 +15,38 @@ export type UseAgentLogsOptions = Readonly<{
   enabled?: boolean;
 }>;
 
+/**
+ * Defines a custom hook that gives you all workspace agent logs for a given
+ * workspace.
+ *
+ * Depending on the status of the workspace, all logs may or may not be
+ * available.
+ */
 export function useAgentLogs(
   options: UseAgentLogsOptions,
 ): readonly WorkspaceAgentLog[] | undefined {
   const { workspaceId, agentId, agentLifeCycleState, enabled = true } = options;
+
   const queryClient = useQueryClient();
   const queryOptions = agentLogs(workspaceId, agentId);
-  const query = useQuery({
-    ...queryOptions,
-    enabled,
-  });
-  const logs = query.data;
+  const query = useQuery({ ...queryOptions, enabled });
+
+  // One pitfall with the current approach: the enabled property does NOT
+  // prevent the useQuery call above from eventually having data. All it does
+  // is prevent it from getting data on its own. Let's say a different useQuery
+  // call elsewhere in the app has the same query key and is enabled. When it
+  // gets data back from the server, the useQuery call here will re-render with
+  // that same new data, even though this state is "disabled". This can EASILY
+  // cause bugs.
+  const logs = enabled ? query.data : undefined;
```

Comment:
This was a fun discovery

Reply:
What kind of bugs? For me, it was expected tho.

Reply:
Basically what happened was:
1. `useAgentLogs` was somewhere in a mounted component on screen, but it was disabled
2. The user clicks the "Download Logs" button, which triggers its own set of queries via `useQueries`
3. One of the query objects from the "Download Logs" functionality shared a query key with `useAgentLogs`, so when it resolved with data, `useAgentLogs` also had its own re-render trigger, even though it was still disabled

Though now that I write that out, maybe it would be okay to revert this. The main thing that tripped me up was that flipping the state variable for the Logs was causing a completely different part of the codebase to error out, but maybe the bulk of the pain would've been removed with stricter compiler settings